addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CrossSiteScriptingScanRule.java

```diff
@@ -684,6 +692,10 @@ public void scan(HttpMessage msg, String param, String value) {
                                                             .getResponseHeader()
                                                             .getHeader(HttpHeader.CONTENT_TYPE),
                                                     "json")) {
+                                                /* avoid sending a null byte to ZAP UI: */
+                                                if (scriptAlert == GENERIC_NULL_BYTE_SCRIPT_ALERT) {
+                                                    scriptAlert = GENERIC_SCRIPT_ALERT;
+                                                }
```

Comment:
- This change is not documented anywhere, not in the commit nor the PR. What's the problem?
 - Should have tests.

Reply:
This is an integral part of the change that is already in the changelog. One of new payloads (which was not used in this place before) contains a null byte. These payloads are then passed as an argument to .setAttack(), which gets then displayed in the UI, so I did not want to break it by putting a null byte there.